### PR TITLE
Matt dev

### PR DESCRIPTION
**Pull request by mstyslinger**
- All crawlers updated for standardization and to include variables previously propagated in post_process.py in the data repo
- Scripts output_spec.json and items.py updated to include additional required variables.
- Tweaks in pipelines.py to account for edits of crawlers

### CODE REVIEW

File: crawler.sh [hunk 1]
```diff
@@ -0,0 +1,11 @@
+export PYTHONPATH="$(pwd)"
```
Comment (takao8) [inline crawler.sh:1]: we should move this file to somewhere into a directory instead of at the root of the repo

File: dataPipelines/gc_scrapy/gc_scrapy/pipelines.py [hunk 2]
```diff
@@ -329,29 +329,29 @@ def process_item(self, item, spider):
...
+            # item["version_hash_raw_data"]["doc_name"] = item["doc_name"]
+            # item["version_hash"] = dict_to_sha256_hex_digest(item["version_hash_raw_data"])
 
         if item.get("access_timestamp") is None:
-            item["access_timestamp"] = datetime.now().strftime("%Y-%m-%d %H:%M:%S.%f")
+            item["access_timestamp"] = datetime.now().strftime("%Y-%m-%dT%H:%M:%S.%f") # T added as delimiter between date and time
 
         if item.get("publication_date") is None:
-            item["publication_date"] = "N/A"
+            item["publication_date"] = None
```
Comment (takao8) [inline dataPipelines/gc_scrapy/gc_scrapy/pipelines.py:341]: we really need to test this. "N/A" works in the pipeline, but None will potentially throw out errors

File: dataPipelines/gc_scrapy/gc_scrapy/pipelines.py [hunk 3]
```diff
@@ -119,7 +119,7 @@ def get_media_requests(self, item, info):
         file_item = self.get_first_supported_downloadable_item(item["downloadable_items"])
 
         if file_item:
-            url = file_item["web_url"]
+            url = file_item["download_url"]
```
Comment (takao8) [inline dataPipelines/gc_scrapy/gc_scrapy/pipelines.py:122]: we need to rigorously check that changing this variable doesn't break our pipeline. this will need to get addressed in the CAC crawlers too

File: dataPipelines/gc_scrapy/gc_scrapy/pipelines.py [hunk 4]
```diff
@@ -329,29 +329,29 @@ def process_item(self, item, spider):
...
         if item.get("publication_date") is None:
-            item["publication_date"] = "N/A"
+            item["publication_date"] = None
 
         if item.get("cac_login_required") is None:
             item["cac_login_required"] = spider.cac_login_required
 
         if item.get("doc_type") is None:
-            item["doc_type"] = getattr(spider, "doc_type", "")
+            item["doc_type"] = getattr(spider, "doc_type", None)
```
Comment (takao8) [inline dataPipelines/gc_scrapy/gc_scrapy/pipelines.py:347]: same as above, check that None won't break the pipeline

File: dataPipelines/gc_scrapy/gc_scrapy/pipelines.py [hunk 5]
```diff
@@ -329,29 +329,29 @@ def process_item(self, item, spider):
...
 
         if item.get("doc_num") is None:
-            item["doc_num"] = ""
+            item["doc_num"] = None
```
Comment (takao8) [inline dataPipelines/gc_scrapy/gc_scrapy/pipelines.py:350]: same as above, check that None won't break the pipeline

File: dataPipelines/gc_scrapy/gc_scrapy/utils.py [hunk 6]
```diff
@@ -42,6 +44,18 @@ def dict_to_sha256_hex_digest(_dict: Dict[Any, Any]) -> str:
 
     return str_to_sha256_hex_digest(value_string)
 
+def get_pub_date(publication_date):
```
Comment (amaruca141) [inline dataPipelines/gc_scrapy/gc_scrapy/utils.py:47]: we need to test to make sure this function works and does not default to "None" automatically

File: dataPipelines/gc_scrapy/gc_scrapy/utils.py [hunk 7]
```diff
@@ -250,3 +264,28 @@ def unzip_docs_as_needed(input_dir: Union[Path, str], output_dir: Union[Path, st
             os.remove(input_dir)
 
     return final_ddocs
+
+
+def parse_timestamp(ts: t.Union[str, datetime.datetime], raise_parse_error: bool = False) -> t.Optional[datetime.datetime]:
```
Comment (amaruca141) [inline dataPipelines/gc_scrapy/gc_scrapy/utils.py:269]: needs to be part of the same review/testing as get_pub_date function

File: dataPipelines/gc_scrapy/gc_scrapy/utils.py (hunk 6 above)
Comment (mstyslinger) [reply dataPipelines/gc_scrapy/gc_scrapy/utils.py:47]: We've tested for all crawlers that utilize this function as part of the "compare analysis" between the dev and matt-dev branches, and the function works as intended

**Review by takao8 — DISMISSED** [review]

**Review by amaruca141 — APPROVED** [review]
